lib/srv/sess.go

```diff
@@ -235,66 +235,59 @@ func (sc *sudoersCloser) Close() error {
 	return nil
 }
 
-func (s *SessionRegistry) TryWriteSudoersFile(ctx *ServerContext) error {
+func (s *SessionRegistry) TryWriteSudoersFile(identityContext IdentityContext) (io.Closer, error) {
```

Comment:
Let's add a godoc here that describes what this method does and what it returns.

In the godoc, it's especially important to explain that `nil, nil` is a valid return (this is atypical, most devs will assume that if the error is nil then the first argument is non-nil).